third_party/nvfuser/csrc/fusion_segmenter.cpp

```diff
@@ -3145,6 +3172,66 @@ void SegmentedFusion::setCachedHeuristicDataFor(
   heuristic_summary_cache_[group] = std::move(data);
 }
 
+void SegmentedFusion::validate(bool require_disjoint) const {
+  // Make sure the groups are a DAG
+  std::unordered_set<SegmentedGroup*> input_groups;
+  std::copy_if(
+      groups().begin(),
+      groups().end(),
+      std::inserter(input_groups, input_groups.end()),
+      [](SegmentedGroup* group) { return group->isInputGroup(); });
+
+  std::unordered_set<SegmentedEdge*> visited_edges;
```

Comment:
Are you emplacing to this set? I don't see it.

Reply:
Oops... Will fix it